src/node_buffer.cc

```diff
@@ -51,19 +67,34 @@ using v8::Number;
 using v8::Object;
 using v8::String;
 using v8::Uint32;
+using v8::Uint8Array;
 using v8::Value;
 
 
 bool HasInstance(Handle<Value> val) {
-  return val->IsObject() && HasInstance(val.As<Object>());
```

Comment:
I might be misreading/missing something obvious, but doesn't `return val->IsObject() && HasInstance(val.As<Object>());` work in both cases?